lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -106,6 +108,22 @@ def write_playbook
         File.write(File.join(@root, 'project', 'playbook.yml'), @playbook)
       end
 
+      def write_ssh_key
+        key_path = File.join(@root, 'env', 'ssh_key')
+        File.symlink(ForemanRemoteExecutionCore.settings[:ssh_identity_key_file], key_path)
+
+        if !@passphrase.nil? && !@passphrase.empty?
+          passwords_path = File.join(@root, 'env', 'passwords')
+          secrets = <<~SECRETS
+            ---
+            "for.*/artifacts/.*/ssh_key_data:": "#{@passphrase}"
+          SECRETS
+          File.write(passwords_path, '')
+          File.chmod(0o0600, passwords_path)
+          File.write(passwords_path, secrets)
```

Comment:
I meant something along the lines of `File.write(passwords_path, secrets, perm: 0o600)` but this should work as well